File: src/primus/pawel/Exercise_153.java
package primus.pawel;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

// Largest pair sum in array
// 7kyu
// https://www.codewars.com/kata/556196a6091a7e7f58000018/train/java
// 2021-06-10 T:21:42:09
class Exercise_153 {
    public static int largestPairSum(int[] numbers) {
        // Stream<Integer>
        List<Integer> list = Arrays.stream(numbers)     // IntStream
                .boxed().sorted().collect(Collectors.toList());

        return list.get(list.size()-1) + list.get(list.size() -2);
    }


}

File: src/JS/Exercise_153.js
// Narcissistic Numbers
// 7kyu
// https://www.codewars.com/kata/56b22765e1007b79f2000079/train/javascript
// 2022-06-29 T:08:40:34

function isNarcissistic(n) {
    return String(n).split("")
        .map(i => Math.pow(i, String(n).length))
        .reduce((num, sum) => num + sum, 0) === n
}

isNarcissistic(153)